### PR TITLE
Redact sensitive header keys from debug logging

### PR DESCRIPTION
**Pull request by bjoerge**

### CODE REVIEW

File: src/middleware/debug.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
 const debugIt = require('debug')
 
+const SENSITIVE_HEADERS = ['Authorization']
```
Comment (rexxars) [inline src/middleware/debug.js]: Should we add `Cookie` to this list?

File: src/middleware/debug.js [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+
+const hasOwn = Object.prototype.hasOwnProperty
+const redactKeys = (source, redactKeys) => {
```
Comment (rexxars) [inline src/middleware/debug.js]: eslint complains about `redactKeys` being shadowed

File: src/middleware/debug.js [hunk 3]
```diff
@@ -29,7 +42,11 @@ module.exports = (opts = {}) => {
       }
 
       if (verbose && options.headers) {
-        log('[%s] Request headers: %s', options.requestId, JSON.stringify(options.headers, null, 2))
+        log(
+          '[%s] Request headers: %s',
+          options.requestId,
+          JSON.stringify(redactKeys(options.headers, SENSITIVE_HEADERS), null, 2)
```
Comment (rexxars) [inline src/middleware/debug.js]: Perhaps we could add a `redactSensitiveHeaders` option that defaults to true? Basically just explicitly check for `opts.redactSensitiveHeaders === false`? At rare occasions you might need to debug something which actually involves the sensitive headers, so having a way to turn that feature off would be great.

File: src/middleware/debug.js (hunk 1 above)
Comment (bjoerge) [reply src/middleware/debug.js]: Sure!

File: src/middleware/debug.js (hunk 3 above)
Comment (bjoerge) [reply src/middleware/debug.js]: Agreed. That's useful

File: src/middleware/debug.js (hunk 2 above)
Comment (bjoerge) [reply src/middleware/debug.js]: Fixed

**Pull request merged**